File: tests/ex-readme.js
try {
   glm = require("../tests/_glm");
} catch(e) {
   load("tests/_glm.js");
}

(function(log) {
    this.mrot = this.mrot || glm.angleAxis(glm.radians(45.0), glm.vec3(0,1,0));
    //...
    var m1 = glm.mat4(1.0); 
    var m2 = glm.mat4(2.0);
    
    var m3 = m1['*'](m2); // operator sugars
    
    m3['*='](glm.toMat4(this.mrot)); // three.js-style inplace mutation still possible

    log("mrot", this.mrot);
    log("m3", m3);

 })(glm.$log)

File: native-tests/ex-readme.cpp
#include "_glm.hpp"





int main() {
    static auto mrot = glm::angleAxis(glm::radians(45.0f), glm::vec3(0,1,0));

    auto m1 = glm::mat4(1.0f); 
    auto m2 = glm::mat4(2.0f);

    auto m3 = m1 * m2;
    log("m3", m3);

    m3 *= glm::toMat4(mrot);

    log("mrot", mrot);
    log("m3", m3);

    return 0;
}
